From b0ccf68f277d0bd5e6fc9d41742f31ddda99a955 Mon Sep 17 00:00:00 2001
From: Keshav Kini <keshav.kini@gmail.com>
Date: Mon, 1 Jun 2020 21:42:24 -0700
Subject: [PATCH 2/2] Restrict RDTSC to x86

Backported from [1].  According to Curtis Dunham, this should fix the ACL2 base
system build on ARM.

[1]: https://github.com/acl2/acl2/commit/292fa2ccc6217e6307d7bb8373eb90f5d258ea5e
---
 memoize-raw.lisp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/memoize-raw.lisp b/memoize-raw.lisp
index 205e78653..478198dee 100644
--- a/memoize-raw.lisp
+++ b/memoize-raw.lisp
@@ -189,7 +189,7 @@
 ;; RDTSC nonsense, but we still can report mysterious results since we have no
 ;; clue about which core we are running on in CCL (or, presumably, SBCL).
 
-#+(or ccl sbcl)
+#+(and (or ccl sbcl) x86-64)
 (eval-when
  (:execute :compile-toplevel :load-toplevel)
  (when #+ccl (fboundp 'ccl::rdtsc)
-- 
2.25.4

